project-koku/koku
# truncate lintitem, lineitem_daily tables

## Description 

Fixes [COST-2650](https://issues.redhat.com/browse/COST-2650)

This is a **CJI script** that will run **multiple sub-processes** each with their own DB connection. The main process will put a task of a table to truncate in each ready worker's task queue. This will continue until all lineitem, lineitem_daily tables have been truncated.

Changes proposed in this PR:
* Truncate (only) the following tables
    * reporting_awscostentrylineitem
    * reporting_awscostentrylineitem_daily
    * reporting_azurecostentrylineitem_daily
    * reporting_gcpcostentrylineitem
    * reporting_gcpcostentrylineitem_daily
    * reporting_ocpusagelineitem
    * reporting_ocpusagelineitem_daily
    * reporting_ocpstoragelineitem
    * reporting_ocpstoragelineitem_daily

## Testing

1. Checkout main
2. `make docker-reinitdb-with-sources`
3. `make load-test-customer-data`
4. Using `psql` do a `select count(*) from acct10001.x;` where `x` is a table out of the above list. Notate the tables with records.
5. Checkout this branch
6. Execute the CJI (in localdev, this can be done from the root dir with `./scripts/cji_scripts/mp_truncate_lineitem_tables.py 4`
7. Repeat step 4. There should no longer be any data in any of those tables (in the `acct10001` schema)
